manifests/config.pp

```diff
@@ -115,118 +113,62 @@
     ensure  => absent,
   }
 
-  if $foreman::apache  {
-    class { 'foreman::config::apache':
-      passenger               => $foreman::passenger,
-      app_root                => $foreman::app_root,
-      passenger_ruby          => $foreman::passenger_ruby,
-      priority                => $foreman::vhost_priority,
-      servername              => $foreman::servername,
-      serveraliases           => $foreman::serveraliases,
-      server_port             => $foreman::server_port,
-      server_ssl_port         => $foreman::server_ssl_port,
-      proxy_backend           => "http://${listen_socket}/",
-      ssl                     => $foreman::ssl,
-      ssl_ca                  => $foreman::server_ssl_ca,
-      ssl_chain               => $foreman::server_ssl_chain,
-      ssl_cert                => $foreman::server_ssl_cert,
-      ssl_certs_dir           => $foreman::server_ssl_certs_dir,
-      ssl_key                 => $foreman::server_ssl_key,
-      ssl_crl                 => $foreman::server_ssl_crl,
-      ssl_protocol            => $foreman::server_ssl_protocol,
-      ssl_verify_client       => $foreman::server_ssl_verify_client,
-      user                    => $foreman::user,
-      passenger_prestart      => $foreman::passenger_prestart,
-      passenger_min_instances => $foreman::passenger_min_instances,
-      passenger_start_timeout => $foreman::passenger_start_timeout,
-      foreman_url             => $foreman::foreman_url,
-      ipa_authentication      => $foreman::ipa_authentication,
-      keycloak                => $foreman::keycloak,
-      keycloak_app_name       => $foreman::keycloak_app_name,
-      keycloak_realm          => $foreman::keycloak_realm,
+  if $foreman::apache and $foreman::ipa_authentication {
+    unless fact('foreman_ipa.default_server') and fact('foreman_ipa.default_realm') {
```

Comment:
Not for this change, but I would have us consider putting external auth into it's own class for slimming down the config class and making it more obvious what aspects are tied to external auth.

Reply:
Probably, there's more room for improvement and I think I scratched the surface here.